src/crimson/osd/osd_operations/internal_client_request.cc

```diff
@@ -65,7 +65,13 @@ seastar::future<> InternalClientRequest::start()
           return enter_stage<interruptor>(
             pp().recover_missing
           ).then_interruptible([this] {
-            return do_recover_missing(pg, get_target_oid());
+            if (pg->is_primary()) {
```

Comment:
I don't think it's possible to have an internal_client_request on a replica -- confirm that, add an assert to the start of the op, and add it to the comment at the top of the class.

Reply:
bb03fc2911fdb378043bb20c9bd6414700fd8a36